vm/isolated/isolated.go

```diff
@@ -175,26 +193,33 @@ func (inst *instance) repair() error {
 	log.Logf(2, "isolated: trying to ssh")
 	if err := inst.waitForSSH(30 * time.Minute); err == nil {
 		if inst.cfg.TargetReboot {
-			log.Logf(2, "isolated: trying to reboot")
-			inst.ssh("reboot") // reboot will return an error, ignore it
-			if err := inst.waitForReboot(5 * 60); err != nil {
-				log.Logf(2, "isolated: machine did not reboot")
-				return err
+			if len(inst.cfg.USBDevNums) > 0 {
+				log.Logf(2, "isolated: trying to reboot by USB authorization")
+				usbAuth := fmt.Sprintf("%s%s%s", "/sys/bus/usb/devices/", inst.cfg.USBDevNums[inst.index], "/authorized")
+				if err := ioutil.WriteFile(usbAuth, []byte("0"), 0); err != nil {
+					log.Logf(2, "isolated: failed to turn off the device")
+					return err
+				}
+				if err := ioutil.WriteFile(usbAuth, []byte("1"), 0); err != nil {
+					log.Logf(2, "isolated: failed to turn on the device")
+					return err
+				}
+			} else {
+				log.Logf(2, "isolated: ssh succeeded, trying to reboot by ssh")
+				inst.ssh("reboot") // reboot will return an error, ignore it
 			}
-			log.Logf(2, "isolated: rebooted wait for comeback")
-			if err := inst.waitForSSH(30 * time.Minute); err != nil {
-				log.Logf(2, "isolated: machine did not comeback")
-				return err
-			}
-			log.Logf(2, "isolated: reboot succeeded")
-		} else {
-			log.Logf(2, "isolated: ssh succeeded")
 		}
-	} else {
-		log.Logf(2, "isolated: ssh failed")
-		return fmt.Errorf("SSH failed")
```

Comment:
Currently we return an error here if ssh failed. With your change we don't have this else branch and always return no error. But we did not actually reboot/repair it and it is probably broken.
Do we need to restore this else branch? Or am I missing something?